app/models/concerns/filterable.rb

```diff
@@ -13,13 +13,13 @@ def filter_scope(name, *args)
       filter_scopes << name
     end
 
-    def filter(filtering_params)
+    def filter_by(filtering_params)
```

Comment:
1. Method name.

Required to change for better support/coverage of cases. While the current version works well for [`User.filter(filtering_params)`](https://github.com/johnmfarrell1/rails_index_filtering/blob/048e67069d5b6fc4cdcae7276af2d4cf16149dde/app/controllers/users_controller.rb#L3) it doesn't work for relations (e.g. `User.all.filter(filtering_params)`). This project is on Rails 6 and Ruby 2.6. Ruby 2.6 added an alias method [`Enumerable#filter`](https://www.bigbinary.com/blog/ruby-2-6-adds-enumerable-filter-as-an-alias-of-enumerable-select) and Rails 5.0+ [includes](https://github.com/rails/rails/commit/b644964b2b555798fc4b94d384b98438db863b3f) `Enumerable` in `ActiveRecord::Relation`. So if you try to filter a relation you'll get `ArgumentError (wrong number of arguments (given 1, expected 0))`.

See this problem mentioned in https://www.justinweiss.com/articles/search-and-filter-rails-models-without-bloating-your-controller/#comment-4463746372.